### DISCUSSION TITLE
vscode配置

### DISCUSSION THREAD

**Discussion by mengchaoheng** [mengchaoheng/SCUT_thesis#6] [General]
这里仅仅简单介绍一下配置，除了配置文件settings.json里设置biber是比较特别的一步以外，其他的都可以参考网上的教程，包括从pdf链接到原代码等等。

可以在settings.json里设置一些自定义配置，比如保存自动编译。

**Comment by mengchaoheng** [comment]
vscode的设置可以参考[知乎](https://zhuanlan.zhihu.com/p/166523064)，当然里面安装texlive部分对于mac用户，是安装[MacTeX](https://tug.org/mactex/mactex-download.html). 像vscode和texstudio这些理解为编辑器/IDE，核心还是tex的发行版。编译时需要调用，编译的流程可以从.json文件和.bat文件看出来。安装好对应版本的发行版之后，我们现在关注vscode的设置：

1.首先下载vscode并安装vscode插件：
<img width="852" alt="161363272-54bd3b6d-02a7-45f6-8c94-5b6fa1123546" src="https://user-images.githubusercontent.com/43166007/168191327-4d09e85b-0f97-4b61-aace-e3b839cbf794.png">



然后选择文件，打开文件夹，选择SCUT_Thesis文件所在文件夹。就打开这个工程了。


2.编译之前需要进行设置，把SCUT_thesis/settings_files/settings.json文件的内容复制，然后按照下面的顺序找到你的配置文件，如果是空白的（一对花括号）就直接覆盖。如果里面有内容，那这个得百度一下，不给一般也简单，只需要把我的内容“加”到你的配置文件中，他的格式是一条一条的，序格式来“加”即可


<img width="1468" alt="截屏2022-04-02 09 41 13" src="https://user-images.githubusercontent.com/43166007/161363460-9778cae6-f1b1-41f9-85f0-6771fba57ac1.png">





3.最后打开一个tex文件scutthesis.tex。就可以编译和查看pdf了：


![截屏2022-04-02 09 56 10](https://user-images.githubusercontent.com/43166007/161363658-92e66727-f298-42fc-bd14-51be2e72e637.png)
点击绿色框 ”Recipe: xelatexmk"进行编译，以后编译点"Build LaTeX project"即可，因为"latex-workshop.latex.recipe.default": "lastUsed"已经设置为选择上一次编译选项。
![截屏2022-07-01 11 27 44](https://user-images.githubusercontent.com/43166007/176817892-9082c26b-0457-4bf4-8df8-19b70fe063c2.png)

**Reply by zhuohoudeputao** [reply]
实际上要用VSCode写latex只需要LaTeX Workshop一个插件就好了，没必要装这么多个吧

**Reply by mengchaoheng** [reply]
是的，latex相关只需要一个插件。git的插件按需安装。

**Reply by mengchaoheng** [reply]
注意还有gitlens ，git graph 的图形界面和gitlens inspect里的比较选项是比较好用的，图形化查看版本更改，比较任意两个版本。

**Comment by mengchaoheng** [comment]
现在的问题是，对于临时文件比如系统文件.DS_Store，还有其他tex编译临时文件，我已经很尽力处理了，但是早期疏忽了，导致历史版本里有这些临时文件，我暂时不知道怎么处理出一条干净 的提交。

**Reply by zhuohoudeputao** [reply]
直接删除.DS_Store然后修改.gitignore添加删除的文件，提交，库上就干净了

**Reply by mengchaoheng** [reply]
> 直接删除.DS_Store然后修改.gitignore添加删除的文件，提交，库上就干净了

有空试一下

**Reply by mengchaoheng** [reply]
> 直接删除.DS_Store然后修改.gitignore添加删除的文件，提交，库上就干净了

已解决

**Comment by mengchaoheng** [comment]
在Mac下，上述问题：对于临时文件比如系统文件.DS_Store，可以删除当前目录的所有.DS_Store，再在.gitignore添加.DS_Store即可解决。我随手百度的几个参考 [全局配置 忽略 .DS_Store](https://blog.csdn.net/weixin_40392053/article/details/122901655)，[如何从Git存储库中删除.DS_Store文件](https://qa.1r1g.com/sf/ask/7539101/)。

**Comment by mengchaoheng** [comment]
git的使用优势就是查看差异和历史提交。latex插件是高亮显示和补齐的。

![截屏2022-04-13 17 52 35](https://user-images.githubusercontent.com/43166007/163153055-90d34186-6f71-4d0e-a4ef-efb210de9fb4.png)

![截屏2022-04-13 17 55 56](https://user-images.githubusercontent.com/43166007/163153347-a6f763d4-7c8f-4301-9ef1-a7029f198f96.png)

**Comment by zhuohoudeputao** [comment]
做了一个简单版的只有latex workshop插件的环境配置视频hihi[https://www.bilibili.com/video/BV1jT4y1e7QT](https://www.bilibili.com/video/BV1jT4y1e7QT)

**Comment by mengchaoheng** [comment]
> 做了一个简单版的只有latex workshop插件的环境配置视频hihi[https://www.bilibili.com/video/BV1jT4y1e7QT](https://www.bilibili.com/video/BV1jT4y1e7QT)

Nice!!!

**Comment by mengchaoheng** [comment]
我最近也有更新，不过不影响论文排版，都只是新加一些文档，还有修正一些地方防止报警。

**Reply by zhuohoudeputao** [reply]
今天发现latexmk是个好东西，一句话就可以编译了
```shell
latexmk -synctex=1 -interaction=nonstopmode -file-line-error -xelatex scutthesis.tex
```

**Comment by mengchaoheng** [comment]
win下：vscode使用ctrl+鼠标左键点击PDF文档跳转到源码，源码到PDF: ctrl+alt+j。
macOS下：vscode使用cmd+鼠标左键点击PDF文档(或直接双击pdf界面)跳转到源码，源码到PDF: cmd+optin+j。

**Reply by mengchaoheng** [reply]
vscode设置里搜索找到Auto Build: Run选项,可将其改为保存自动编译。

**Comment by mengchaoheng** [comment]
针对编辑界面文字超出范围不显示的问题，如下图：
![截屏2022-05-13 09 29 04](https://user-images.githubusercontent.com/43166007/168193373-297082c9-9d79-4005-ba83-e766d42965cd.png)


想设置为超出范围的文字自动换行（不是最终论文换行，只是显示换行），点击vs code的左下角齿轮->选择”设置“，找到常用设置里的“Editor: Word Wrap”，选“on"。

![截屏2022-05-13 09 22 50](https://user-images.githubusercontent.com/43166007/168193276-e30e90b9-83f5-4ecc-9e1e-481e06c83693.png)

设置后如下图：




![截屏2022-05-13 09 31 36](https://user-images.githubusercontent.com/43166007/168193582-04e0f180-5825-4897-b14a-cb417a4a33ba.png)


参考：
[VScode如何自动换行设置](https://wenku.baidu.com/view/87228681f624ccbff121dd36a32d7375a417c633.html)
